py/desispec/spectra.py

```diff
@@ -658,6 +666,168 @@ def update(self, other):
+                meta['extra_catalog'] = self.extra_catalog
+            sl.append(Spectrum1D(flux=flux, spectral_axis=spectral_axis,
+                                 uncertainty=uncertainty, mask=mask, meta=meta))
+        return sl
+
+    @classmethod
+    def from_specutils(cls, spectra):
+        """Convert ``specutils`` objects to a :class:`~desiutil.spectra.Spectra` object.
+
+        Parameters
+        ----------
+        spectra : specutils.Spectrum1D or specutils.SpectrumList
+            A ``specutils`` object.
+
+        Returns
+        -------
+        :class:`~desiutil.spectra.Spectra`
+            The corresponding DESI-internal object.
+
+        Raises
+        ------
+        NameError
+            If ``specutils`` is not available in the environment.
+        ValueError
+            If an unknown type is found in `spectra`.
+        """
+        if not _specutils_imported:
+            raise NameError("specutils is not available in the environment.")
+        if isinstance(spectra, SpectrumList):
+            sl = spectra
+            try:
+                bands = sl[0].meta['bands']
+            except KeyError:
+                #
+                # This is a big assumption; it doesn't capture ['b', 'z'] or ['r', 'z'].
+                #
+                bands = ['b', 'r', 'z'][0:len(sl)]
+        elif isinstance(spectra, Spectrum1D):
+            sl = [spectra]
+            #
+            # Assume this is a coadd across cameras.
+            #
+            try:
+                bands = sl[0].meta['bands']
+            except KeyError:
+                bands = ['brz']
+        else:
+            raise ValueError("Unknown type input to Spectra.from_specutils!")
+        #
+        # Load objects that are independent of band from the first item.
+        #
+        try:
+            fibermap = sl[0].meta['fibermap']
+        except KeyError:
+            fibermap = None
+        try:
+            exp_fibermap = sl[0].meta['exp_fibermap']
+        except KeyError:
+            exp_fibermap = None
+        try:
+            meta = sl[0].meta['desi_meta']
+        except KeyError:
+            meta = None
+        try:
+            single = sl[0].meta['single']
+        except KeyError:
+            single = False
+        try:
+            scores = sl[0].meta['scores']
+        except KeyError:
+            scores = None
+        try:
+            scores_comments = sl[0].meta['scores_comments']
+        except KeyError:
+            scores_comments = None
+        try:
+            extra_catalog = sl[0].meta['extra_catalog']
+        except KeyError:
+            extra_catalog = None
+        #
+        # Load band-dependent quantities.
+        #
+        wave = dict()
+        flux = dict()
+        ivar = dict()
+        mask = dict()
+        resolution_data = None
+        extra = None
+        for i, band in enumerate(bands):
+            wave[band] = sl[i].spectral_axis.value
+            flux[band] = sl[i].flux.value
```

Comment:
as long as these are unit-ful items that you have to extract the non-unit values, consider explicitly converting them to `AA` and `specunit` before taking `.value`, just in case they are provided in some other units.

Reply:
Good point, I think that's easy.

Reply:
Done now.